arch/arm/src/stm32h7/hardware/stm32h7x3xx_pwr.h

```diff
@@ -123,7 +123,11 @@
 
 #define STM32_PWR_CR3_BYPASS        (1 << 0)  /* Bit 0: Power management unit bypass */
 #define STM32_PWR_CR3_LDOEN         (1 << 1)  /* Bit 1: Low drop-out regulator enable */
+#if defined(CONFIG_STM32H7_STM32H7X7XX)
+#define STM32_PWR_CR3_SMPSEN        (1 << 2)  /* Bit 2: Switched-mode power supply enable */
```

Comment:
Being concerned about if def-rash, it this the correct approach? 

1) Are there enough differences  that this needs it's own include file? 
2) Should it be feature based not IC?

I.E. CONFIG_STM32H7_HAS_SMPS 

Then is it is shareable as the feature level and any chip can add that define and avoid code churn listing every IC.

Reply:
I am actually just following the style under stm32h7. All other places seem to be using CONFIG_STM32H7_STM32H7X7XX , there are no feature select mentions in the KConfig.
For single core operations all STM32H7XX chips have same registers and configurations down to the pin config except for PWR selection, there is an additional option to select SMPS, and its either LDO or SMPS. SMPS/LDO PWR Selection feature is part of all dual core offerings by ST, as better power regulator is needed for Dual Core operations which can't be fulfilled by LDO.

The clean way of doing this would probably require a much larger change, like changing the file names of all stm32h7x3xx headers to a stm32h7xxxx common header with defines wrt dual core feature, and adding support for dual core operations etc. Maybe until that time comes this is a good stopgap solution?

Reply:
>  Maybe until that time comes this is a good stopgap solution?

Not really. With very little effort you can add STM32H7_HAS_SMPS to the Kconfig or chip.h and do this the correct way. If not this will grow with every change into the mess stm32 is. 

Please make the change upstream and we can then back port it.

Reply:
@davids5 I'm trying to get this upstream:

https://github.com/apache/nuttx/pull/8124

Your review would be very much appreciated. I hope I'm on the right track there.

Reply:
@julianoes Sorry I was OOO.